### PR TITLE
add perl-Tk

### PR DESCRIPTION
**Pull request by shen390s**
add support of perl-Tk to nixos

### CODE REVIEW

**Comment by shen390s** [comment]
some nixes package(such as texdoctk in texlive) need perl-tk, this pull request add perl-tk to nixes packages

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -10485,4 +10485,27 @@ let self = _self // overrides; _self = with self; {
...
+
+    buildInputs = [ pkgs.xlibs.libXext pkgs.xlibs.libX11 pkgs.libpng ];
+
+    configurePhase=''
+        perl Makefile.PL PREFIX=$out X11LIB=${pkgs.xlibs.libX11}/lib X11INC=${pkgs.xlibs.libX11}/include 
+    '';
+
+    NIX_CFLAGS_LINK = "-lpng";
+
+    doCheck = false;
```
Comment (pSub) [inline pkgs/top-level/perl-packages.nix]: Why do you disable the tests?

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -10485,4 +10485,27 @@ let self = _self // overrides; _self = with self; {
...
+
+    meta = {
+      description = "Tk - a Graphical User Interface Toolkit";
```
Comment (pSub) [inline pkgs/top-level/perl-packages.nix]: `description` should not repeat the package name (see the nixpkgs manual).

File: pkgs/top-level/perl-packages.nix (hunk 1 above)
Comment (shen390s) [inline pkgs/top-level/perl-packages.nix]: after test, it seems that we can delete the line "doCheck = false".

File: pkgs/top-level/perl-packages.nix (hunk 2 above)
Comment (shen390s) [inline pkgs/top-level/perl-packages.nix]: Can you suggest the text for description?

**Comment by shen390s** [comment]
change description to:
description = "A Graphical User Interface Toolkit";

**Comment by pSub** [comment]
Thanks. Could you please squash your commits with a message like "Tk: init at 804.033".

**Comment by shen390s** [comment]
commit message changed to "Tk: init at 804.033"

**Comment by jagajaga** [comment]
Rebase please. Leave just one commit.

**Comment by shen390s** [comment]
rebase nixos/release-14.12

**Comment by shen390s** [comment]
merge commits into single commit with message "Tk: init at 804.033"

**Pull request merged**